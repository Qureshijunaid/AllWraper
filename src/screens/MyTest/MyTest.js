
import React,{useState} from 'react'
import { Button, Text, TextInput, View } from 'react-native'
function MyTest() {
    const [name, setUser] = React.useState('')
    const [age,setAge]=useState('0')
    const [show, setShow] = React.useState(false)
    return (
        <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
            <TextInput
                value={name}
                onChangeText={setUser} testID="input"
                style={{ height: 100, width: 100, borderWidth: 2, borderColor: 'black' }}
            />
            <TextInput
                value={age}
                keyboardType="numeric"
                onChangeText={setAge} testID="inputAge"
                style={{ height: 100, width: 100, borderWidth: 2, borderColor: 'black' }}
            />
            <Button
                title="Print Username"
                onPress={() => {
                    setTimeout(() => {
                        setShow(!show)
                    }, Math.floor(Math.random() * 200))
                }}
            />
            {show && <Text testID="printed-username">{age}</Text>}
        </View>
    )
}

export default MyTest



