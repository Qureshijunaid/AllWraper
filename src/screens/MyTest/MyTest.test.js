import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import MyTest from './MyTest';

test('examples of some things', async () => {
    const { getByTestId, getByText, queryByTestId, toJSON } = render(<MyTest />)
    const famousProgrammerInHistory = 'Ada Lovelace'
    const middleAge = '23'

    const input = getByTestId('input')
    //   console.log("input in test==>",input)
    fireEvent.changeText(input, famousProgrammerInHistory)
    const inputAge = getByTestId('inputAge')

    fireEvent.changeText(inputAge, middleAge)

    const button = getByText('Print Username')
    // console.log("button==>", button)
    fireEvent.press(button)


    await waitFor(() => expect(queryByTestId('printed-username')).toBeTruthy())

    expect(getByTestId('printed-username').props.children).toBe(
        middleAge
    )
    expect(toJSON()).toMatchSnapshot()
});


