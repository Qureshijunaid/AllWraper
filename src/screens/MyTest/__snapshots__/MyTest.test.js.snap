// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 2,
        "height": 100,
        "width": 100,
      }
    }
    testID="input"
    underlineColorAndroid="transparent"
    value="Ada Lovelace"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 2,
        "height": 100,
        "width": 100,
      }
    }
    testID="inputAge"
    underlineColorAndroid="transparent"
    value="23"
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Print Username
      </Text>
    </View>
  </View>
  <Text
    testID="printed-username"
  >
    23
  </Text>
</View>
`;
